test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,95 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new ConsoleEnvironment();
+            string dataPath = "iris.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, ctx => (
+               label: ctx.LoadText(0),
+               sepalWidth: ctx.LoadFloat(1),
+               sepalLength: ctx.LoadFloat(2),
+               petalWidth: ctx.LoadFloat(3),
+               petalLength: ctx.LoadFloat(4)),
+               dataPath);
+
+
+            var pipeline = data.MakeEstimator()
+                .Append(row => (
+                    // Convert string label to key.
+                    label: row.label.Dictionarize(),
+                    // Concatenate all features into a vector.
+                    features: row.sepalWidth.ConcatWith(row.sepalLength, row.petalWidth, row.petalLength)))
+                .Append(row => (
+                    label: row.label,
+                    prediction: row.label.PredictWithMultiClassFastTree(row.features)));
+
+            // Load the data into the system.
+            var data = dataReader.Read(dataPath);
+            var model = pipeline.Fit(data);
```

Comment:
This example doesn't gain much from having `dataReader.Read(dataPath)`, right? Would it make sense for `pipeline.Fit()` to take a `dataPath`? This would be more similar to Scikit-Learn, but perhaps is not in sync with other concepts here.

Reply:
if your model is fitted using a data path, then it can only work from another data path (and not individual examples, for example)

---
In reply to: [215141509](https://github.com/dotnet/machinelearning/pull/809#discussion_r215141509) [](ancestors = 215141509)